Some "url" attributes causes cmake-conan to fail

Having some special characters in the URL attribute in a conanfile.py may cause `conan_provider.cmake`  to fail (see error message below). In my case it was a "?" and a "=" that generated this error. Selected generators was CMakeDeps but dont think it matters

```
CMake Error at /srv/rolwal/repos/sync-product/cmake/conan_provider.cmake:476 (string):
  string sub-command JSON failed parsing json string: * Line 1492, Column 24

    Syntax error: value, object or array expected.

  .
Call Stack (most recent call first):
  /srv/rolwal/repos/sync-product/cmake/conan_provider.cmake:578 (conan_install)
  CMakeLists.txt:60 (find_package)

```

--- a/conan_provider.cmake
+++ b/conan_provider.cmake
@@ -473,13 +473,13 @@ function(conan_install)
     # the files are generated in a folder that depends on the layout used, if
     # one is specified, but we don't know a priori where this is.
     # TODO: this can be made more robust if Conan can provide this in the json output
-    string(JSON CONAN_GENERATORS_FOLDER GET ${conan_stdout} graph nodes 0 generators_folder)
+    string(JSON CONAN_GENERATORS_FOLDER GET "${conan_stdout}" graph nodes 0 generators_folder)
     cmake_path(CONVERT ${CONAN_GENERATORS_FOLDER} TO_CMAKE_PATH_LIST CONAN_GENERATORS_FOLDER)
     # message("conan stdout: ${conan_stdout}")
     message(STATUS "CMake-Conan: CONAN_GENERATORS_FOLDER=${CONAN_GENERATORS_FOLDER}")
     set_property(GLOBAL PROPERTY CONAN_GENERATORS_FOLDER "${CONAN_GENERATORS_FOLDER}")
     # reconfigure on conanfile changes
-    string(JSON CONANFILE GET ${conan_stdout} graph nodes 0 label)
+    string(JSON CONANFILE GET "${conan_stdout}" graph nodes 0 label)
     message(STATUS "CMake-Conan: CONANFILE=${CMAKE_SOURCE_DIR}/${CONANFILE}")
     set_property(DIRECTORY ${CMAKE_SOURCE_DIR} APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS "${CMAKE_SOURCE_DIR}/${CONANFILE}")
     # success

--- a/example/CMakeLists.txt
+++ b/example/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.24)
 
-project(FormatOutput LANGUAGES CXX)
+project(FormatOutput LANGUAGES CXX C)
 
 set(CMAKE_CXX_STANDARD 14)
 

--- a/example/conanfile.txt
+++ b/example/conanfile.txt
@@ -3,3 +3,6 @@ fmt/9.1.0
 
 [layout]
 cmake_layout
+
+[generators]
+CMakeDeps

